mono/metadata/mempool.c

```diff
@@ -442,10 +442,8 @@ mono_mempool_foreach_block(MonoMemPool* pool, mono_mempool_block_proc callback,
 
 	while (current)
 	{
-		gpointer start = (guint8*)current + SIZEOF_MEM_POOL;
 		gpointer end = (guint8*)current + current->size;
-
-		callback(start, end, user_data);
+		callback(current, end, user_data);
```

Comment:
changing the chunk iteration to this would ensue that the user sees correct allocation ranges